src/modules/widgets/saga.ts

```diff
@@ -468,18 +494,44 @@ export function* editChartWidget({
   }
 }
 
+export function* editImageWidget({
+  payload,
+}: ReturnType<typeof editImageWidgetAction>) {
+  const { id } = payload;
+
+  const state = yield select();
+  const widgetId = getWidget(state, id).widget.id;
+
+  yield put(showImagePicker());
+  const action = yield take([SAVE_IMAGE]);
```

Comment:
In my opinion - here we need to add a `HIDE_IMAGE_PICKER ` constant as user could also cancel the image edit flow.